src/python/pants/jvm/resolve/jvm_tool.py

```diff
@@ -0,0 +1,314 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+import importlib.resources
+import json
+import logging
+from dataclasses import dataclass
+from typing import ClassVar, Iterable, Sequence, cast
+
+from pants.backend.python.target_types import UnrecognizedResolveNamesError
+from pants.engine.fs import CreateDigest, Digest, FileContent, MergeDigests, Workspace
+from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.internals.selectors import Get, MultiGet
+from pants.engine.rules import collect_rules, goal_rule, rule
+from pants.engine.unions import UnionMembership, union
+from pants.jvm.resolve.coursier_fetch import (
+    ArtifactRequirements,
+    Coordinate,
+    CoursierResolvedLockfile,
+)
+from pants.option.subsystem import Subsystem
+from pants.util.logging import LogLevel
+from pants.util.ordered_set import FrozenOrderedSet
+
+DEFAULT_TOOL_LOCKFILE = "<default>"
```

Comment:
Now that we're doing this in two places, we should probably extract a config constants file and have `python_tool_base` draw from it as well.

Reply:
I think this should probably go in `core/goals/generate_lockfiles.py` once we unify the langauges